include/pisa/util/inverted_index_utils.hpp

```diff
@@ -12,12 +12,12 @@ void emit(std::ostream &os, const uint32_t *vals, size_t n)
 
 void emit(std::ostream &os, uint32_t val) { emit(os, &val, 1); }
 
-
 // sample_fn must be stable, it must return a sorted vector
 template <typename SampleFn>
 void sample_inverted_index(std::string const &input_basename,
                            std::string const &output_basename,
-                           SampleFn &&sample_fn)
+                           SampleFn &&sample_fn,
```

Comment:
Why not use a `std::function` instead a template?

Reply:
I'm good with either, we're using `std::function` anyway, so might as well put it in the signature.